### ISSUE TITLE
Multiscrape seems to have mixed up the service names and adding a name to a sensor creates a service which does not trigger anything

### ISSUE THREAD

**Issue by jwinterton** [danieldotnl/ha-multiscrape#181]
I checked other issues - can only see one other issue logged here.

Daniel - thanks for Multiscrape, it is really useful.
I think this may be a bug but apologies if not. All the logging appears to be geared towards interpreting responses from scraping. Thats not what I have a problem with so the logging was not a great help - however if you need anymore I will happily provide it.
 
Relevant background:
I have an un-named Multiscrape sensor configured. This works very well. I did not need or use the trigger service for this use case. I only require this sensor during the Summer so at the moment the whole configuration is #'d out in the config.yaml.

Current issue:
I configured a new (initially) un-named sensor for a new use case. In this case I wanted to use the trigger service. When I went to look for the trigger service I see that there are two named:
Multiscrape scraping component: trigger_scraper_noname_0
Multiscrape scraping component: trigger_scraper_noname_1

Initially I did not put two and two together to associate _0 with the #'d out sensor config. So I was confused as to why I had two services and not sure which to use. So to identify the correct service I named the new senor "HW Tank".
This created a third service called:
Multiscrape scraping component: trigger_hw_tank

Now I observe the following behaviour:
When I call Multiscrape scraping component: trigger_scraper_noname_0
The new sensor is polled (I can see the HTML request arriving on sensor device - an esp8266) which responds but the sensors in HA are not being updated.
When I call Multiscrape scraping component: trigger_scraper_noname_1
The new sensor is polled AND the sensors in HA are being correctly updated. (so it all works as expected)

When I call Multiscrape scraping component: trigger_hw_tank
Nothing happens - no poll of the device.

So I think the bug is - the services are not cleaned up when the sensor is removed from config (I assume they should be ??) the orphaned service seems to be triggering another sensor definition but not updating the sensors. The properly named service is not working at all possibly since there is a default service name already associated with the sensor.

These are snippets of the relevant code:
Original (currently disabled sensor)
```
# multiscrape:
  # - resource: http://10.x.x.75/ping
    # scan_interval: 1800
    # sensor:
      # - unique_id: hydro_ph
        # name: pH
        # select: "title"
        # value_template: '{{ (value.split(";")[1]) }}'
        # unit_of_measurement: "pH"
```
etc ...

New sensor
```
multiscrape:
  - name: HW Tank
  - resource: http://10.x.x.66/readings
    scan_interval: 300
    sensor:
      - unique_id: hw0_tempC
        name: Hot Water Tank Probe 0
        select: "title"
        value_template: '{{ (value.split(";")[0]) }}'
        unit_of_measurement: "C"
```
etc ...

Multiscrape: v6.6.1
HA Core: 2022.12.8

**Comment by danieldotnl** [comment]
Thank you for the extensive issue description. I think I can reproduce this by renaming a multiscrape config and reloading multiscrape from the UI. It indeed result in 2 different services which is incorrect.
However, restarting home assistant solves this. Is that correct in your case as well?

**Comment by jwinterton** [comment]
OK that is odd. I have restarted many times since I #'d out the old sensor - not least right after #'ing it out to remove it from the running config!

However - having just rebooted to answer your question I do see something different. noname_0 has gone. I just have noname_1 and hw_tank.
Before:
![image](https://user-images.githubusercontent.com/44443807/215887976-fac27892-eb2b-4aaa-80f1-de4471d7e30b.png)
Then I restarted HA from the UI.
After restart:
![image](https://user-images.githubusercontent.com/44443807/215891061-7a6e75a1-3d4d-4741-877c-1493f219b376.png)

So I think the answer is no it doesn't correct it in the way you are seeing. I assume I should only see hw_tank?

Please let me know if I can do anything else.

**Comment by danieldotnl** [comment]
Could you share your full multiscrape config?

**Comment by jwinterton** [comment]
Sure no problem. This is the excerpt from config.yaml:

```
multiscrape:
  - name: HW Tank
  - resource: http://10.x.x.66/readings
    scan_interval: 300
    sensor:
      - unique_id: hw0_tempC
        name: Hot Water Tank Probe 0
        select: "title"
        value_template: '{{ (value.split(";")[0]) }}'
        unit_of_measurement: "C"
      - unique_id: hw1_tempC
        name: Hot Water Tank Probe 1
        select: "title"
        value_template: '{{ (value.split(";")[1]) }}'
        unit_of_measurement: "C"
      - unique_id: hw2_tempC
        name: Hot Water Tank Probe 2
        select: "title"
        value_template: '{{ (value.split(";")[2]) }}'
        unit_of_measurement: "C"
      - unique_id: hw3_tempC
        name: Hot Water Tank Probe 3
        select: "title"
        value_template: '{{ (value.split(";")[3]) }}'
        unit_of_measurement: "C"
      - unique_id: hw4_tempC
        name: Hot Water Tank Probe 4
        select: "title"
        value_template: '{{ (value.split(";")[4]) }}'
        unit_of_measurement: "C"
      - unique_id: hw5_tempC
        name: Hot Water Tank Probe 5
        select: "title"
        value_template: '{{ (value.split(";")[5]) }}'
        unit_of_measurement: "C"
      - unique_id: hw_inlet_tempC
        name: Hot Water Tank Inlet
        select: "title"
        value_template: '{{ (value.split(";")[6]) }}'
        unit_of_measurement: "C"
      - unique_id: hw_outlet_tempC
        name: Hot Water Tank Outlet
        select: "title"
        value_template: '{{ (value.split(";")[7]) }}'
        unit_of_measurement: "C"
# multiscrape:
  # - resource: http://10.x.x.75/ping
    # scan_interval: 1800
    # sensor:
      # - unique_id: hydro_ph
        # name: pH
        # select: "title"
        # value_template: '{{ (value.split(";")[1]) }}'
        # unit_of_measurement: "pH"
      # - unique_id: hydro_ec
        # name: EC
        # select: "title"
        # value_template: '{{ (value.split(";")[2]) }}'
        # unit_of_measurement: "µS/cm"
      # - unique_id: hydro_do2
        # name: DO2
        # select: "title"
        # value_template: '{{ (value.split(";")[3]) }}'
        # unit_of_measurement: "mg/L"
      # - unique_id: hydro_water_temp
        # name: Water Temp
        # select: "title"
        # value_template: '{{ (value.split(";")[4]) }}'
        # unit_of_measurement: "°C"
      # - unique_id: hydro_greenhouse_humidity
        # name: Humidity
        # select: "title"
        # value_template: '{{ (value.split(";")[5]) }}'
        # device_class: "humidity"
        # unit_of_measurement: "%"
      # - unique_id: hydro_air_temp
        # name: Air Temp
        # select: "title"
        # value_template: '{{ (value.split(";")[6]) }}'
        # unit_of_measurement: "°C"
      # - unique_id: hydro_light_level
        # name: LUX
        # select: "title"
        # value_template: '{{ (value.split(";")[8]) }}'
        # device_class: "illuminance"
        # unit_of_measurement: "Lx"
      # - unique_id: hydro_fills
        # name: Fills
        # select: "title"
        # value_template: '{{ (value.split(";")[12]) }}'
      # - unique_id: hydro_fill_butt_volume
        # name: Fill Butt Volume
        # select: "title"
        # value_template: '{{ (value.split(";")[13]) | round | int }}'
        # unit_of_measurement: "litres"
      # - unique_id: hydro_nutrients
        # name: Total Nutrients
        # select: "title"
        # value_template: '{{ (value.split(";")[14]) }}'
        # unit_of_measurement: "mls"
      # - unique_id: hydro_ph_up
        # name: pH UP
        # select: "title"
        # value_template: '{{ (value.split(";")[15]) }}'
        # unit_of_measurement: "mls"
      # - unique_id: hydro_ph_down
        # name: pH DOWN
        # select: "title"
        # value_template: '{{ (value.split(";")[16]) }}'
        # unit_of_measurement: "mls"
      # - unique_id: hydro_nuta_level
        # name: Nutrient A Level
        # select: "title"
        # value_template: '{{ (value.split(";")[17]) }}'
        # unit_of_measurement: "mm"
      # - unique_id: hydro_nutb_level
        # name: Nutrient B Level
        # select: "title"
        # value_template: '{{ (value.split(";")[18]) }}'
        # unit_of_measurement: "mm"
    # binary_sensor:
      # - unique_id: hydro_pump
        # name: Pump Status
        # select: "title"
        # value_template: >-
          # {%- if (value.split(";")[11]) == "1" -%}
            # On
          # {%- elif (value.split(";")[11]) == "0" -%}
            # Off
          # {%- endif -%}
```

**Comment by roumano** [comment]
> Sure no problem. This is the excerpt from config.yaml:
> 
> ```
> multiscrape:
>   - name: HW Tank
>   - resource: http://10.x.x.66/readings
>     scan_interval: 300
>     sensor:
>       - unique_id: hw0_tempC
>         name: Hot Water Tank Probe 0
>         select: "title"
>         value_template: '{{ (value.split(";")[0]) }}'
>         unit_of_measurement: "C"
>       - unique_id: hw1_tempC
>         name: Hot Water Tank Probe 1
>         select: "title"
>         value_template: '{{ (value.split(";")[1]) }}'
>         unit_of_measurement: "C"
>       - unique_id: hw2_tempC
>         name: Hot Water Tank Probe 2
>         select: "title"
>         value_template: '{{ (value.split(";")[2]) }}'
>         unit_of_measurement: "C"
>       - unique_id: hw3_tempC
>         name: Hot Water Tank Probe 3
>         select: "title"
>         value_template: '{{ (value.split(";")[3]) }}'
>         unit_of_measurement: "C"
>       - unique_id: hw4_tempC
>         name: Hot Water Tank Probe 4
>         select: "title"
>         value_template: '{{ (value.split(";")[4]) }}'
>         unit_of_measurement: "C"
>       - unique_id: hw5_tempC
>         name: Hot Water Tank Probe 5
>         select: "title"
>         value_template: '{{ (value.split(";")[5]) }}'
>         unit_of_measurement: "C"
>       - unique_id: hw_inlet_tempC
>         name: Hot Water Tank Inlet
>         select: "title"
>         value_template: '{{ (value.split(";")[6]) }}'
>         unit_of_measurement: "C"
>       - unique_id: hw_outlet_tempC
>         name: Hot Water Tank Outlet
>         select: "title"
>         value_template: '{{ (value.split(";")[7]) }}'
>         unit_of_measurement: "C"
> # multiscrape:
>   # - resource: http://10.x.x.75/ping
>     # scan_interval: 1800
>     # sensor:
>       # - unique_id: hydro_ph
>         # name: pH
>         # select: "title"
>         # value_template: '{{ (value.split(";")[1]) }}'
>         # unit_of_measurement: "pH"
>       # - unique_id: hydro_ec
>         # name: EC
>         # select: "title"
>         # value_template: '{{ (value.split(";")[2]) }}'
>         # unit_of_measurement: "µS/cm"
>       # - unique_id: hydro_do2
>         # name: DO2
>         # select: "title"
>         # value_template: '{{ (value.split(";")[3]) }}'
>         # unit_of_measurement: "mg/L"
>       # - unique_id: hydro_water_temp
>         # name: Water Temp
>         # select: "title"
>         # value_template: '{{ (value.split(";")[4]) }}'
>         # unit_of_measurement: "°C"
>       # - unique_id: hydro_greenhouse_humidity
>         # name: Humidity
>         # select: "title"
>         # value_template: '{{ (value.split(";")[5]) }}'
>         # device_class: "humidity"
>         # unit_of_measurement: "%"
>       # - unique_id: hydro_air_temp
>         # name: Air Temp
>         # select: "title"
>         # value_template: '{{ (value.split(";")[6]) }}'
>         # unit_of_measurement: "°C"
>       # - unique_id: hydro_light_level
>         # name: LUX
>         # select: "title"
>         # value_template: '{{ (value.split(";")[8]) }}'
>         # device_class: "illuminance"
>         # unit_of_measurement: "Lx"
>       # - unique_id: hydro_fills
>         # name: Fills
>         # select: "title"
>         # value_template: '{{ (value.split(";")[12]) }}'
>       # - unique_id: hydro_fill_butt_volume
>         # name: Fill Butt Volume
>         # select: "title"
>         # value_template: '{{ (value.split(";")[13]) | round | int }}'
>         # unit_of_measurement: "litres"
>       # - unique_id: hydro_nutrients
>         # name: Total Nutrients
>         # select: "title"
>         # value_template: '{{ (value.split(";")[14]) }}'
>         # unit_of_measurement: "mls"
>       # - unique_id: hydro_ph_up
>         # name: pH UP
>         # select: "title"
>         # value_template: '{{ (value.split(";")[15]) }}'
>         # unit_of_measurement: "mls"
>       # - unique_id: hydro_ph_down
>         # name: pH DOWN
>         # select: "title"
>         # value_template: '{{ (value.split(";")[16]) }}'
>         # unit_of_measurement: "mls"
>       # - unique_id: hydro_nuta_level
>         # name: Nutrient A Level
>         # select: "title"
>         # value_template: '{{ (value.split(";")[17]) }}'
>         # unit_of_measurement: "mm"
>       # - unique_id: hydro_nutb_level
>         # name: Nutrient B Level
>         # select: "title"
>         # value_template: '{{ (value.split(";")[18]) }}'
>         # unit_of_measurement: "mm"
>     # binary_sensor:
>       # - unique_id: hydro_pump
>         # name: Pump Status
>         # select: "title"
>         # value_template: >-
>           # {%- if (value.split(";")[11]) == "1" -%}
>             # On
>           # {%- elif (value.split(";")[11]) == "0" -%}
>             # Off
>           # {%- endif -%}
> ```

For me, it's look likz a issue on your code, you have a extra `-` before your resource

So it's should be : 
```
multiscrape:
  - name: HW Tank
    resource: http://10.x.x.66/readings
```

**Comment by jwinterton** [comment]
That is a good spot! I removed it and reloaded the Multiscrape config. Both multiscrape services: noname_1 and hw_tank were still there but hw_tank now works as expected! 
I rebooted HA and it tidied up noname_1 as you suggested.

That is a peculiar error as it kind of worked but didn't. I guess I had not noticed it before because I have not previously used the service and it didn't seem to affect the other functionality.

 Thanks for your help Daniel - much appreciated.
